---
title: "Metavenom Project"
author: "Agneesh Barua"
date: "25/02/2019"
output: html_document
---

```{r, libraries, warning=FALSE, echo=FALSE}
library(tidyverse)
library(edgeR)
library(WGCNA)
library(DESeq2)
library(GOstats)
library(GSEABase)
library(GO.db)
library(reshape2)
library(RSQLite)
library(AnnotationDbi)
library(flashClust)
library(ade4)

```

```{r, warning=FALSE}
options(stringsAsFactors = F)
enableWGCNAThreads()
```

```{r, datasets}

rsem <- read_csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/Pm Network/data/rsem.csv.gz", col_types = "ccnncnccc")
ercc <- read_csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/Pm Network/data/ercc.csv.gz", col_types = cols())
factors <- read_csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/Pm Network/data/factors.csv", col_types = cols()) %>% filter(tissue == "venom gland")
```

## Quality control
ERCC92 was added to the RNA-seq libraries for quality checking
```{r qc}
fmt <- function() function(x) as.character(round(x,2))
ggplot(ercc, aes(conc, tpm, color=library, group=library))+geom_point()+scale_x_continuous(labels = fmt(),trans="log2")+ scale_y_continuous(labels = fmt(),trans="log2")+theme_bw()+ xlab("Spike-in concentration (attamol/ul)")+ylab("Spike-in TPM") + geom_hline(yintercept = .05, color = "red")+guides(color=F)
```

There is generally good log-linear response between observed and expected values, though we can't detect anything lower than 0.05 TPM. Looks like one library got less spike in than the rest.

```{r}

rsemVenom <- rsem[grepl('Pm_\\d+$', rsem$library),]
# remove low expressed loci
keep <- rsemVenom %>% group_by(gene) %>% summarize(tpm = mean(tpm)) %>% filter(tpm>0.05) %>% dplyr::select(gene) # genes to keep
tpmVenom <- rsemVenom %>% dplyr::select(library, gene, count, description, class) %>% filter(gene %in% keep$gene) %>%  spread(key = library, value = count)  %>%  remove_rownames %>% column_to_rownames("gene")
# do differential gene expression on venom components only
countsVenom <- rsemVenom %>% filter(!is.na(class)) %>% dplyr::select(library, gene, count) %>%  spread(key = library, value = count) %>% 
  remove_rownames %>% column_to_rownames("gene")
#design matrix
design <- model.matrix(~factor(day)+factor(sex)+0, factors)
rownames(design) <- rownames(factors)
dge <- DGEList(countsVenom)
dge <- calcNormFactors(dge)
dat <- estimateGLMCommonDisp(dge, design)
dat <- estimateGLMTrendedDisp(dat, design)
dat <- estimateGLMTagwiseDisp(dat, design)
fit <- glmFit(dat,design)
mds <- plotMDS(dge)
mdsPlot <- data.frame(mds1=mds$cmdscale.out[,1], mds2=mds$cmdscale.out[,2], ID=rownames(mds$cmdscale.out),days=factors$day,sex=factors$sex,mass=factors$mass)
ggplot(mdsPlot,aes(mds1,mds2,size=mass,shape=sex,color=as.factor(days)))+geom_point()+scale_colour_grey(start=0.8, end = 0)
maleVsfemale <- glmLRT(fit,coef=5)  
summary(caste.de <- decideTestsDGE(maleVsfemale, p=0.05, adjust="BH"))
ggplot(na.omit(subset(rsem, tissue == "venom gland")), aes(class,tpm, group=sex)) + scale_y_log10()+ stat_summary(fun.data = "mean_cl_boot", position = position_dodge(width = .75), aes(colour = sex))
t.test(mass~sex,factors)
t.test(svl~sex,factors)
```

#Gene network analysis
```{r}
datExpr <- t(rlog(as.matrix(round(tpmVenom%>%dplyr::select(-c(Pm_7, description,class))))))

snakesampleTree<-hclust(dist(datExpr), method = "average") #clustering samples as a dendrogram, based on a certain measure to see if there are any outliers. 
plot(snakesampleTree)
#No outliers keeping all tips

#Network construction
powers = c(seq(from = 10, to=16, by=1))

#call network topology analyis function
allowWGCNAThreads()
sft <- pickSoftThreshold(datExpr, powerVector = powers, verbose = 5, networkType = "signed")

sizeGrWindow(9, 5)
par(mfrow = c(1,2))
cex1 = 0.9

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"))
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red")

# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red") 

plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")
```

```{r}
genes_coexpr_venom<-read.csv("./WGCNA/Pm Network/data/genes_coexpresed_with_venom.genes.csv")$gene
tpmVenom_mtest <- tpmVenom %>% rownames_to_column("gene") %>% dplyr::filter(gene %in% genes_coexpr_venom) %>%  remove_rownames %>% column_to_rownames("gene")
  
datExpr_mtest <- t(rlog(as.matrix(round(tpmVenom_mtest%>%dplyr::select(-c(Pm_7, description,class))))))

snakesampleTree<-hclust(dist(datExpr_mtest), method = "average") #clustering samples as a dendrogram, based on a certain measure to see if there are any outliers. 

sizeGrWindow(9, 5)
par(mfrow = c(1,2))
cex1 = 0.9

#Network construction
powers = c(seq(from = 1, to=20, by=1))

#call network topology analyis function
allowWGCNAThreads()
sft <- pickSoftThreshold(datExpr_mtest, powerVector = powers, verbose = 5, networkType = "signed")

# Scale-free topology fit index as a function of the soft-thresholding power
plot(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
     main = paste("Scale independence"))
text(sft$fitIndices[,1], -sign(sft$fitIndices[,3])*sft$fitIndices[,2],
     labels=powers,cex=cex1,col="red")

# this line corresponds to using an R^2 cut-off of h
abline(h=0.90,col="red") 

plot(sft$fitIndices[,1], sft$fitIndices[,5],
     xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
     main = paste("Mean connectivity"))
text(sft$fitIndices[,1], sft$fitIndices[,5], labels=powers, cex=cex1,col="red")

#calculate adjacency
adjacency<-adjacency(datExpr_mtest,3)

data_cor<-dist(as.matrix(cor(datExpr_mtest)))
adja_cor<-dist(as.matrix(adjacency))

mantel.rtest(data_cor,adja_cor,nrepet = 999)

a <- rnorm(100)
a
a_ad<-cor(a, (0.5 * (1+a) )^2)

a_cor<-


```

We'll choose soft thresholding power 13, as the smallest that crosses the 0.9 threshold.
```{r}
#Co-expression similarity and adjacency
softPower <- 13
adjacency <- adjacency(datExpr ,power = softPower)
```

Topological overlap matrix (TOM)
To minimize effects of noise and spurious associations, we transform the adjacency into Topological Overlap Matrix, 
and calculate the corresponding dissimilarity:
```{r, eval=FALSE}
TOM <- TOMsimilarity(result$adjacency)
dissTOM <- 1-TOM

#Clustering using TOM. Hierarchial clustering to produece hierarchial clustering tree (dendrogram) of genes. 
geneTree <- hclust(as.dist(dissTOM), method = 'average')
plot(geneTree, xlab="", sub = "", main = "Gene clustering on TOM-based dissimilarity", labels = F, hang = 0.04)
```

```{r}
TOM<-readRDS("./WGCNA/RDS_files/venom_TOM.rds")
dissTOM<-readRDS("./WGCNA/RDS_files/venom_dissTOM.rds")

```

```{r,include=FALSE}
geneTree<-readRDS("./WGCNA/RDS_files/venom_geneTree.rds")
plot(geneTree, xlab="", sub = "", main = "Gene clustering on TOM-based dissimilarity", labels = F, hang = 0.04)
```
Module identification using dynamic tree cut:
```{r, eval=FALSE}
minModuleSize = 30
dynamicMods <- cutreeDynamic(dendro = geneTree, distM = 1-TOM, deepSplit = 2, pamRespectsDendro = FALSE, minClusterSize = minModuleSize);
table(dynamicMods)
```

```{r, include=FALSE}
dynamicMods<-readRDS("./WGCNA/RDS_files/venom_dynamicMods.rds")
table(dynamicMods)
```

```{r}
#Convert numeric labels into colours
dynamicColours <- labels2colors(dynamicMods)
table(dynamicColours)

#plot dendrogram with colours
sizeGrWindow(8,6)
plotDendroAndColors(geneTree, dynamicColours, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colors")
```

Merging modules whose expression profiles are very similar. The dynamics tree can sometimes identify modules whose expression profiles are very similar. 
It may  be prudent to merge such modules since their genes are highly co-expressed. To quantify co-expression similarity of entire modules, we
calculate their eigengenes and cluster them on their correlation:

```{r}
#Calculate eigengenes:
MEList <- moduleEigengenes(datExpr, colors = dynamicColours)
MEs <- MEList$eigengenes

#Calculate dissimilarity of module eigengenes
MEDiss <- 1-cor(MEs)

#Cluster module eigengenes
METree <- hclust(as.dist(MEDiss), method = "average")

#Plot the result
sizeGrWindow(7,6)
plot(METree, main = "Clustering of module eigengenes",
     xlab = "", sub = "")
#A height cut off of 0.25, corresponds to a correlation of 0.75 to merge.
MEDissThres = 0.2
# Plot the cut line into the dendrogram
abline(h=MEDissThres, col = "red")
```

```{r}
# Call an automatic merging function
merge = mergeCloseModules(datExpr, dynamicColours, cutHeight = MEDissThres, verbose = 3)
# The merged module colors
mergedColors = merge$colors;
# Eigengenes of the new merged modules:
mergedMEs = merge$newMEs

#plot
plotDendroAndColors(geneTree, cbind(dynamicColours, mergedColors),
                    c("Dynamic Tree Cut", "Merged dynamic"),
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05)


# Rename to moduleColors
moduleColors = mergedColors
# Construct numerical labels corresponding to the colors
colorOrder = c("grey", standardColors(50));
moduleLabels = match(moduleColors, colorOrder)-1;
MEs = mergedMEs;

# Recalculate MEs with color labels
invisible(MEs0 <- moduleEigengenes(datExpr, moduleColors)$eigengenes)
MEs = orderMEs(MEs0)

print(table(moduleColors))
moduleColors <- as.data.frame(moduleColors)
rownames(moduleColors) <- colnames(datExpr)

```

Save module colors and labels for use in subsequent parts
```{r}
save(MEs, moduleLabels, moduleColors, geneTree, file = "./WGCNA/SnakeNetwork_Md_res.RData")
```

The above results of the WGCNA were also saved as an RDS file, which we will use in further downstream analysis. 
## Which modules contain venom genes
```{r}
result<-readRDS("./WGCNA/RDS_files/wgcna.results.rds")
tpmVenom$colors <- result[["moduleColors"]]$moduleColors
genes.and.their.modules <- na.omit(tpmVenom) %>% dplyr::select(description, colors) #frame with genes and their modules in the same order as input
```

Finding which module contains venom genes
```{r}
venom.desc <-read.csv("./WGCNA/Pm Network/data/venom_genes_desc.csv") #has descriptions of venom genes that we'll search for
filter(genes.and.their.modules, grepl(paste(venom.desc$description, collapse = "|"),description)) #Searches for descriptions of venom.desc (with separator '|') in genes.with..
```

Therefore, most venom genes are in module turquoise. Now we isolate all the genes in turquoise module. (or the genes conexpressed with venom genes)
```{r}
#module with genes of interest
genes_colors <- as.data.frame(c(result$moduleColors, keep))
genes_coexpressed_with_venom<-filter(genes_colors, moduleColors == "turquoise")
genes_coexpressed_with_venom
```

Save dataset with all genes in turquoise module
```{r, eval=F}
write.csv(genes_coexpressed_with_venom, file = "genes_coexpresed_with_venom.genes.csv")
```

Calculate topological overlap anew: this could be done more efficiently by saving the TOM calculated during module detection, but we do it again here.
```{r}
dissTOM = 1-TOMsimilarityFromExpr(datExpr, power = 13)
# Transform dissTOM with a power to make moderately strong connections more visible in the heatmap
plotTOM = dissTOM^7
# Set diagonal to NA for a nicer plot
diag(plotTOM) = NA
# Call the plot function
result$moduleColors
sizeGrWindow(9,9)
TOMplot(plotTOM, geneTree, tpmVenom$colors, main = "Network heatmap plot, all genes")
```













#Module preservation
```{r}
tpmLiver <- rsem[grepl('liver', rsem$library),] %>% dplyr::select(library, gene, count) %>% filter(gene %in% keep$gene) %>% spread(key = library, value = count)  %>%  remove_rownames %>% column_to_rownames("gene")

datExprLiver <- t(rlog(as.matrix(round(tpmLiver))))
colnames(datExprLiver) <- rownames(tpmLiver)
dim(datExprLiver)

gsg<-goodSamplesGenes(datExprLiver, verbose = 3)
gsg$allOK #if returns TRUE, all genes have passed the cut. If not, remove the offending genes and samples from the data.

if (!gsg$allOK)
{
  # Optionally, print the gene and sample names that were removed:
  if (sum(!gsg$goodGenes)>0)
    printFlush(paste("Removing genes:", paste(names(datExprLiver)[!gsg$goodGenes], collapse = ", ")));
  if (sum(!gsg$goodSamples)>0)
    printFlush(paste("Removing samples:", paste(rownames(datExprLiver)[!gsg$goodSamples], collapse = ", ")));
  # Remove the offending genes and samples from the data:
  datExprLiver = datExprLiver[gsg$goodSamples, gsg$goodGenes]
}
```

```{r, eval=FALSE}
allowWGCNAThreads()
dissTOM.Liver<-1-TOMsimilarityFromExpr(datExprLiver, power = 13)
```

```{r, include=FALSE}
dissTOM.Liver<-readRDS("./WGCNA/RDS_files/Liver_dissTOM.Liver.rds")
```

```{r}
tree.Liver<-flashClust(as.dist(dissTOM.Liver), method = "a")
#plot(tree.Liver, xlab="", sub = "", main = "Gene clustering on TOM-based dissimilarity", labels = F, hang = 0.04)

minModuleSize = 30
dynamicMods.Liver <- cutreeDynamic(dendro = tree.Liver, distM = dissTOM.Liver, deepSplit = 2, pamRespectsDendro = FALSE, minClusterSize = minModuleSize);

#Convert numeric labels into colours
dynamicColours.Liver <- labels2colors(dynamicMods.Liver)
table(dynamicColours.Liver)

#plot dendrogram with colours
sizeGrWindow(8,6)
plotDendroAndColors(tree.Liver, dynamicColours.Liver, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colors")


layout(matrix(c(1:4), 4, 1), heights = rep(c(0.8, 0.2), 2));
#venom
plotDendroAndColors(geneTree, dynamicColours, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colors")
#liver
plotDendroAndColors(tree.Liver, dynamicColours.Liver, "Dynamic Tree Cut",
                    dendroLabels = FALSE, hang = 0.03,
                    addGuide = TRUE, guideHang = 0.05,
                    main = "Gene dendrogram and module colors")
```



```{r}
setLabels = c("venom", "liver");
multiExpr = list(venom = list(data = datExpr), liver = list(data = datExprLiver));
multiColor = list(venom = dynamicColours, liver = dynamicColours.Liver);
nSets = 2

mp = modulePreservation(multiExpr, multiColor,
referenceNetworks = c(1:2),
nPermutations = 200,
randomSeed = 1,
verbose = 3)

saveRDS(mp,"./WGCNA/RDS_files/mp_liver.rds")

```


#GO term enrichment
GO terms were obtained using Blast2GO.
```{r}
#Preparing the data
Pm_data<-read.csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/Pm_GO_clean2.csv")


Pm_melt <- Pm_data %>%
  gather(gene.id, GO.ids, X:X.239) %>%
  arrange(Gene.id)

Pm_melt2<-Pm_melt[!(is.na(Pm_melt$GO.ids) | Pm_melt$GO.ids==""),]

#write.csv(Pm_melt2, file = "/Users/agneesh-barua/Dropbox (OIST)/Pm_GO terms/Pm_GO_List.csv")
```

```{r, warning=FALSE}
#Enrichment with GOstats

#Getting genes from WGCNA results

##Preparing the GO frame
annot<-read.csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/Pm Network/Seq data/Pm_GO_List.csv") 
genes_with_venom<-read.csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/ATF6_transcriptome_analysis/csv_files/genes_coexpresed_with_venom.genes.csv")

annot<-annot %>% mutate(evidence = "IEA")  %>% dplyr::select(go_id= GO.ids, evidence, gene=Gene_id)

goFrame<-GOFrame(as.data.frame( annot, organism = "Pm"))
goAllFrame<-GOAllFrame(goFrame)

gsc <-GeneSetCollection(goAllFrame, setType = GOCollection())

universe <- unique(annot$gene)
mode(universe)

genes<- as.data.frame(genes_with_venom$gene)
intersect(universe,genes)
NROW(universe)
NROW(genes)
```

GO term enrichment step
```{r, warning=FALSE}
params <- GSEAGOHyperGParams(name = "Pm GO enrich", 
                             geneSetCollection = gsc,
                             geneIds = genes,
                             universeGeneIds = universe,
                             ontology = "BP",
                             pvalueCutoff = 0.05,
                             conditional = F,
                             testDirection = "over")


Over<- hyperGTest(params)
summary(Over) %>% arrange(Pvalue) %>% head(50)
#GO_enrich<-as.data.frame(summary(Over))
#GO_enrich %>% arrange(Pvalue) %>% write.csv(file = "GO_term_enrichment_Pm.red.module.csv")
```

#Differential Expression analysis
```{r}
# formatting the data

rsem<-read.csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/WGCNA/Pm Network/data/rsem.csv")

long_data<-rsem %>% filter(day == '1') %>% filter(!is.na(class)) %>% dplyr::select(library, gene, count, description) %>%
  dplyr::arrange(library) %>% spread(library, count) %>% remove_rownames %>% column_to_rownames("gene")
```

We select venom gland, liver, and kidneys for differential expression analysis. We select lib 1,3 and 5.
```{r,include=FALSE}
long_data$Pm_4<-NULL
long_data$Pm_6<-NULL
long_data$Pm_2<-NULL
long_data$Pm_1_heart<-NULL
long_data$Pm_3_heart<-NULL
long_data$Pm_5_heart<-NULL
gene_desc<-long_data$description
long_data$description<-NULL
#long_data$Pm_4<-NULL
#long_data$Pm_6<-NULL
```

```{r}
# DGE object
groups<-factor(c("venom",'kidney','liver',"venom",'kidney','liver',"venom",'kidney' ,'liver'))
dge<-DGEList(long_data, group = groups, genes = gene_desc)
levels(dge$samples$group)
dge
```

Keeps genes with cpm > 1, and if present in two or more libraries
```{r}
keep<- filterByExpr(dge)                 #rowSums(cpm(dge)>1) >= 2
summary(keep)
dge<- dge[keep,,keep.lib.sizes = F]
```

We normalise using TMM: Done for removing compositional bias
```{r}
dge<-calcNormFactors(dge, method = "TMM")
dge$samples
plotMD(cpm(dge, log = T),column =1)
abline(h=0, col="red", lty=2, lwd=2)

plotMD(cpm(dge, log = T),column =2)
abline(h=0, col="red", lty=2, lwd=2)
#ideally, the bulk of genes should be centered around zero, construct for each sample by replacing column number
plotMD(cpm(dge, log = T),column =3)
abline(h=0, col="red", lty=2, lwd=2)

plotMD(cpm(dge, log = T),column =4)
abline(h=0, col="red", lty=2, lwd=2)

plotMD(cpm(dge, log = T),column =5)
abline(h=0, col="red", lty=2, lwd=2)

plotMD(cpm(dge, log = T),column =6)
abline(h=0, col="red", lty=2, lwd=2)

plotMD(cpm(dge, log = T),column =7)
abline(h=0, col="red", lty=2, lwd=2)

plotMD(cpm(dge, log = T),column =8)
abline(h=0, col="red", lty=2, lwd=2)
```

Examine inter sample relationship by doing an MDS plot
```{r}
plotMDS(dge)
```

To estimate dispersions with multiple factors we donstruct a design matrix. 0+ means not to include an intercept
```{r}
design <- model.matrix(~0+group, dge$samples)
colnames(design)<-levels(dge$samples$group)
design
```
Setting up the DE
```{r}
#common dispersion
dge<-estimateDisp(dge, design, robust = T)
dge$common.dispersion
plotBCV(dge)

#determining quasi liklihood 
fit<-glmQLFit(dge, design, robust = T)
head(fit$coefficients)
plotQLDisp(fit)
```

##Differential expression analysis
```{r}
con<-makeContrasts(venom-liver, levels = design)
qlf<- glmQLFTest(fit,contrast=con)
topTags(qlf,100)

summary(decideTests(qlf))
plotMD(qlf)
```

Get genes that have a log fold change of 1.2
```{r}
tr<-glmTreat(fit,contrast = con, lfc = log2(1.5))
(topTags(tr,100))
summary(decideTests(tr))
plotMD(tr)
```

##ANOVA-like testing
```{r}
con2<-makeContrasts( +venom-liver,
                     +venom-kidney,
                    levels = design)

anov<-glmQLFTest(fit, contrast = con2)

gene_id<-rownames(anov$genes)
desc<-anov$genes$genes
anov$table %>% mutate(genes=gene_id, description=desc) %>% arrange(-logFC..venom...liver,logFC..venom...kidney) #%>% write.csv("DE_anova.csv")
```

##UPR and ERAD genes upregulated in the venom gland.
To get the list on genes associated with UPR and ERAD we simply search the genes which have the GO:BP terms for UPR and ERAD.
```{r, warnings=FALSE}
GO_file<-read.csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/Pm Network/Seq data/Pm_GO_List.csv")

GO_terms<-read_csv("/Users/agneesh-barua/Dropbox (OIST)/R Training/Meta-venom/Pm_GO terms/GO_term_enrichment_Pm.csv")
```

We selected the following 11 GO terms:
```{r}
GO_of_interest<-c('GO:0034976', #response to endoplasmic reticulum stress
                  'GO:0006986', #response to unfolded protein
                  'GO:0034620', #cellular response to unfolded protein
                  'GO:0035966', #response to topologically incorrect protein
                  'GO:0035967', #cellular response to topologically incorrect protein
                  'GO:0030968', #endoplasmic reticulum unfolded protein response
                  'GO:0048193', #Golgi vesicle transport
                  'GO:0071702', #organic substance transport
                  'GO:0006520', #cellular amino acid metabolic process
                  'GO:0019752', #carboxylic acid metabolic process
                  'GO:0036503') #ERAD pathway

GO_terms %>% dplyr::select(GOBPID, Term) %>% filter(GOBPID %in% GO_of_interest) #%>% write.csv("Terms.csv")
```

Select all the genes with above associated GO terms.
```{r}
Gene_of_interest<-GO_file %>% dplyr::select(Gene_id,GO.ids) %>% filter(GO.ids %in% GO_of_interest)
genes_i_want<-Gene_of_interest$Gene_id
```

Find those genes in the anova:
```{r}
DE_anov<-read.csv("./ATF6_transcriptome_analysis/csv_files/DE_anova.csv")

DE_anov %>% filter(genes %in% genes_i_want) %>% arrange(-logFC..venom...kidney, -logFC..venom...liver)# %>% write.csv("Genes_in_ERAD_UPR.csv")
```

Find specific gene of interest (optional)
```{r}
genes_of_interest<-c(107287284,107302856,107288132,107290253,107294279,107292378,107285759,107290253,107291672,107294454,107297064,107292140,107284391,
                     107293306,
                     107285595,
                     107296774,
                     107294617,
                     107285132,
                     107284495,
                     107290978,
                     107286499,
                     107289787,
                     107294779,
                     107292484,
                     107285739,
                     107289378,
                     107284078)
anov$table %>% mutate(genes=gene_id, description=desc) %>% arrange(-logFC..venom...liver,logFC..venom...kidney) %>% filter(genes %in% genes_of_interest) %>% view# write.csv("DE_anova_genes_of_intrst.csv")
```
